pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -171,7 +172,7 @@ static SqlNodeAndOptions extractSqlNodeAndOptions(String sql, SqlNodeList sqlNod
     if (sqlType == null) {
       throw new SqlCompilationException("SqlNode with executable statement not found!");
     }
-    return new SqlNodeAndOptions(statementNode, sqlType, options);
+    return new SqlNodeAndOptions(statementNode, sqlType, QueryOptionsUtils.resolveCaseInsensitiveOptions(options));
```

Comment:
this has several issues
1. using legacy OPTION keyword will still be case-sensitive; and legacy OPTIONS keyword is written into the map after resolving SqlNodeAndOptions; how do we resolve conflicts in this situation? for example:
```
SET useMultistageengine = true;
SELECT ...
OPTION(useMultistageEngine, false)
```
2. SqlNodeAndOption only parses options in query, how does it handle json payload option keys?

Reply:
I considered both of these and thought it was OK to make this a partial improvement. the SET syntax is particularly misleading given that it "feels" like sql - if we feel strongly about it I can attempt to wire this in through the other places as well.

Reply:
it should be fine. but let's make sure to test when the other 2 ways to configure options are present - and what the desired behavior is - and that should match the behavior of current logic.

Reply:
e.g. also test
1. JSON option `timeoutMs = 60000`
2. query option `OPTION(timeoutMs=30000)`
3. query set `SET timeoutMs = 20000`
4. implicitly there's a default 10000

let's make sure the combination of these result in the same final timeout number
- by setting 1,2,3 to `timeoutms` or `TIMEOUTMS` or `timeoutMs`

Reply:
looks like there was actually a pretty easy way to make all of them work :) I'll update the PR

Reply:
confirmed with the following:
```
select * from dates limit 10; #no multistage default
select * from dates limit 10 OPTION(usemultistageengine=true) #use options
SET usemultistageengine=true; select * from dates limit 10;
```
and also via curl for json:
```
curl 'http://localhost:9000/sql' --data-raw '{"sql":"select * from dates limit 10","trace":false,"queryOptions":"usemultistageengine=true"}'
```